File: laravel-backend/app/Models/Vehicle.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Vehicle extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'owner_id',
        'vehicle_type_id',
        'registration_number',
        'chassis_number',
        'engine_number',
        'brand',
        'model',
        'manufacture_year',
        'color',
        'seat_count',
        'first_registration_date',
        'purpose_of_use',
        'registration_status',
        'status'
    ];

    protected $dates = ['first_registration_date'];

    public function owner()
    {
        return $this->belongsTo(Owner::class);
    }

    public function vehicleType()
    {
        return $this->belongsTo(VehicleType::class);
    }

    public function registrationRequests()
    {
        return $this->hasMany(RegistrationRequest::class);
    }

    public function inspections()
    {
        return $this->hasMany(Inspection::class);
    }

    public function appointments()
    {
        return $this->hasMany(Appointment::class);
    }

    public function latestInspection()
    {
        return $this->hasOne(Inspection::class)->latestOfMany();
    }

    public function latestPassedInspection()
    {
        return $this->hasOne(Inspection::class)->whereIn('result', ['passed', 'conditional'])->latest();
    }


    // Scopes
    public function scopeRegistered($query)
    {
        return $query->where('registration_status', 'registered');
    }

    public function scopeActive($query)
    {
        return $query->where('status', 'active');
    }

    // Helpers
    public function needsInspection()
    {
        if (!$this->latestInspection) {
            return true;
        }

        $nextInspectionDate = $this->latestInspection->next_inspection_date;
        return now()->greaterThanOrEqualTo($nextInspectionDate);
    }
}

File: android-frontend/app/src/main/java/com/example/frontend2/domain/model/Vehicle.kt
package com.example.frontend2.domain.model

data class Vehicle(
    val id: Int,
    val vehicleTypeId: Int? = null,
    val vehicleTypeName: String? = null,
    val registrationNumber: String,
    val chassisNumber: String? = null,
    val engineNumber: String? = null,
    val brand: String,
    val model: String? = null,
    val manufactureYear: Int? = null,
    val color: String,
    val seatCount: Int? = null,
    val firstRegistrationDate: String? = null,
    val purposeOfUse: String? = null,
    val registrationStatus: String? = null,
    val status: String? = null,
    val createdAt: String? = null,
    val updatedAt: String? = null
)

data class VehicleErrorState(
    var errorVehicleTypeId: String = "",
    var errorRegistrationNumber: String = "",
    var errorChassisNumber: String = "",
    var errorEngineNumber: String = "",
    var errorBrand: String = "",
    var errorModel: String = "",
    var errorManufactureYear: String = "",
    var errorColor: String = "",
    var errorSeatCount: String = "",
    var errorFirstRegistrationDate: String = "",
    var errorPurposeOfUse: String = "",
)